cmd/oras/internal/option/confirmation.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package option
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/spf13/pflag"
+)
+
+// Confirmation option struct.
+type Confirmation struct {
+	Confirmed bool
+}
+
+// ApplyFlags applies flags to a command flag set.
+func (opts *Confirmation) ApplyFlags(fs *pflag.FlagSet) {
+	fs.BoolVarP(&opts.Confirmed, "yes", "y", false, "do not prompt for confirmation")
+}
+
+func (opts *Confirmation) AskForComfirmation(message string) (bool, error) {
```

Comment:
Forgot to add unit test for this func, will do.

Reply:
A reference for [how `az acr manifest delete` do user confirmation](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/acr/manifest.py#L649), [step into it](https://github.com/Azure/azure-cli/blob/6fbdd00e735934305b853061b1bc966a154012fb/scripts/curl_install_pypi/install.py#L88).